clang-tools-extra/test/clang-tidy/checkers/bugprone/tagged-union-member-count.cpp

```diff
@@ -0,0 +1,181 @@
+// RUN: %check_clang_tidy %s bugprone-tagged-union-member-count %t
```

Comment:
The default config is `-std=c++11-or-later`, but this check is not limited to that only, so we want to run it with `-std=c++98-or-later`.

The check also supports C and Objective-C, so please add `.c` and `.m` test files and create tests for those languages too. 

If you don't want to support any of these languages, please add `bool isLanguageVersionSupported(const LangOptions &LangOpts) const override;` to your check and configure the supported languages.